publisher.go

```diff
@@ -193,6 +196,13 @@ func (p *RetryPublisher) Publish(ctx context.Context, exchange, key string, msg
 	return err
 }
 
+// RetryAttempt returns the current retry attempt of a message published by RetryPublisher.
+// retryAttempt, ok := RetryAttempt(ctx)
```

Comment:
leftover comment? 

maybe worth mentioning the func will return false as 2nd arg if the key do not exist in the ctx